subsys/net/ip/net_if.c

```diff
@@ -350,8 +354,9 @@ void net_if_queue_tx(struct net_if *iface, struct net_pkt *pkt)
 	if ((IS_ENABLED(CONFIG_NET_TC_SKIP_FOR_HIGH_PRIO) &&
 	     prio >= NET_PRIORITY_CA) || NET_TC_TX_COUNT == 0) {
 		net_pkt_set_tx_stats_tick(pkt, k_cycle_get_32());
-
+		net_if_lock(iface);
```

Comment:
Wouldn't it make sense to lock the mutex before stats update?

Reply:
Yes

Reply:
I meant, the stats update above, in line 346.

Reply:
Ah ok. But wouldn't that make this block till the TX is complete? I want to keep the queueing of TX packets not wait on actual TX completion. Even limiting the lock to stats update and not the actual queueing would block the TX in case other thread is calling `net_process_tx_packet`?

Reply:
If that's a concern, then in typical data flow this would be the case regardless of whether we lock in `net_if_queue_tx()`, as UDP/TCP data path calls `net_if_send_data()` which takes the lock anyway.

Reply:
ok, let's leave that for now, will move the lock to the entire function.